examples/lib/stories/camera_and_viewport/coordinate_systems.dart

```diff
@@ -0,0 +1,232 @@
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flame/palette.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+/// A simple widget that "wraps" a Flame game with some Contaienrs
+/// on each direction (top, bottom, left and right) and allow adding
+/// or removing containers.
+class CoordinateSystemsWidget extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() {
+    return _CoordinateSystemsState();
+  }
+}
+
+class _CoordinateSystemsState extends State<CoordinateSystemsWidget> {
+  /// The number of blocks in each direction (top, left, right, bottom).
+  List<int> blocks = [1, 1, 1, 1];
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        ...createBlocks(index: 0, rotated: false, start: true),
+        Expanded(
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              ...createBlocks(index: 1, rotated: true, start: true),
+              Expanded(
+                child: GameWidget(game: CoordinateSystemsGame()),
+              ),
+              ...createBlocks(index: 2, rotated: true, start: false),
+            ],
+          ),
+        ),
+        ...createBlocks(index: 3, rotated: false, start: false),
+      ],
+    );
+  }
+
+  /// Just creates a list of Widgets + the "add" button
+  List<Widget> createBlocks({
```

Comment:
note to reviewer: this sadly cannot be a Widget because it returns a list of Widgets

Reply:
I can't see exactly how that could be implemented in a nice way just by your description, but we could pair on it in a followup if you are up for it :)

Reply:
I will merge this for now and lmk if you want to help me refactor we can do a followup.